elixir/apps/domain/lib/domain/accounts/account/changeset.ex

```diff
@@ -2,22 +2,61 @@ defmodule Domain.Accounts.Account.Changeset do
   use Domain, :changeset
   alias Domain.Accounts.Account
 
-  def create_changeset(attrs) do
-    %Account{}
+  def changeset(account, attrs) do
+    account
     |> cast(attrs, [:name, :slug])
     |> validate_required([:name])
-    |> validate_length(:name, min: 1, max: 255)
+    |> validate_name()
+    |> trim_name_whitespace()
+    |> set_slug_default()
+    |> downcase_slug()
+    |> remove_spaces_from_slug()
     |> validate_slug()
-    |> validate_length(:slug, min: 3, max: 100)
+    |> unique_constraint(:slug, name: :accounts_slug_index)
+  end
+
+  def create_changeset(attrs) do
+    %Account{}
+    |> changeset(attrs)
+  end
+
+  defp validate_name(changeset) do
+    changeset
+    |> validate_length(:name, min: 3, max: 255)
+    |> validate_format(:name, ~r/^[a-zA-Z0-9\s_-]+$/,
+      message: "can only contain letters, numbers, spaces, underscores, and dashes"
+    )
+  end
+
+  defp trim_name_whitespace(changeset) do
+    update_change(changeset, :name, &String.trim/1)
+  end
+
+  defp set_slug_default(changeset) do
```

Comment:
I would keep it as `put` just to align with the naming convention (`Map.put/3`, `Ecto.Changeset.put_change/3` etc):
```suggestion
  defp put_slug_default(changeset) do
```

Reply:
👍 makes sense

Reply:
This has been updated.